### PR TITLE
Scripts for setup dev environment for clu cmdlet work

### PR DESCRIPTION
**Pull request by yugangw-msft**
1. "SetupEnv.bat" invoke "build-local" and prepares everything to execute cmdlets. Takes about 4 minutes. In detail, it includes
   a. Build source projects
   b. Build clu cmdlet packages and deploy to local command repo 
   c. Update and deploy msclu.cfg
   d. Install all cmdlet packages 
   e. Add "<repo-root>\drop\clurun\win7-x64" to %path%, so you can run clu commands.
   f. Turn on "DebugCLU", so you can attach debuggers
   g. It warns if dotnet.exe is not under PATH
2. "BuildCmdlet.bat", let you build individual cmdlet. Take about 20 seconds.
   For example: BuildCmdlet.bat Microsoft.Azure.Commands.Profile

EDIT: push the new commits which prepare osx & linux bits to xcopy over and execute

### CODE REVIEW

File: tools/CLU/BuildDrop.ps1 [hunk 1]
```diff
@@ -9,9 +9,6 @@ if (!($workspaceDirectory))
     $env:WORKSPACE = $workspaceDirectory
 }
 
-$buildProfileScriptPath = "`"$thisScriptDirectory\BuildProfile.ps1`"" # Guard against spaces in the path
-$sourcesRoot = "$workspaceDirectory\src\clu"
```
Comment (yugangw-msft) [inline tools/CLU/BuildDrop.ps1]: i move this after the folder creating script lines, because buildprofile.ps1 appears want to ensure the folder existence.

**Comment by markcowl** [comment]
@johanste the clu changes here look fine to me, but can you review?

@yugangw-msft We should work together with Johan to add in a runtime parameter to the build scripts, so we can build out osx and Ubuntu cmdlet packages as well.

File: tools/CLU/SetupEnv.bat [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+echo ^(Get-Content "%mscluCfg%"^) ^| ForEach-Object { $_ -replace "TOFILL", "%root%\drop\CommandRepo" } ^| Set-Content "%mscluCfg%"^ >"%temp%\Rep.ps1"
+@powershell -file %temp%\Rep.ps1
+
+set DebugCLU=
+%root%\drop\clurun\win7-x64\clurun.exe --install
+%root%\drop\clurun\win7-x64\clurun.exe --install Microsoft.Azure.Commands.Profile
+%root%\drop\clurun\win7-x64\clurun.exe --install Microsoft.Azure.Commands.Resources
+%root%\drop\clurun\win7-x64\clurun.exe --install Microsoft.Azure.Commands.Websites
+%root%\drop\clurun\win7-x64\clurun.exe --install Microsoft.Azure.Commands.Management.Storage
```
Comment (hovsepm) [inline tools/CLU/SetupEnv.bat]: also add Microsoft.Azure.Commands.Resources.Cmdlets . I'm finalizing a PR to produce that package as well.

**Comment by yugangw-msft** [comment]
@johanste @markcowl I pushed a new commit with the script which prepares osx and linux bits. The 2 folders can xcopy to linux/osx box and execute there, which i have verified in ubuntu 14.04 Once nuget client fixes the bug, the script can avoid propagating 'pkgs' folder; rather should use "clurun --install" to get all set up

**Comment by yugangw-msft** [comment]
reviewed by hovsep

**Pull request merged**